### PR TITLE
FBS: Consumer

### PR DESCRIPTION
**Pull request by jmillan**
Move Rust Consumer related communication to FBS.

### CODE REVIEW

**Comment by jmillan** [comment]
@nazar-pc, could you please take a look at the Rust side of this and [Producer PR](https://github.com/versatica/mediasoup/pull/1164) whenever you have the time? Thanks!

This PR takes the former as the base. I did it this way no to get blocked until the former was reviewed, but I don't want to stack more PRs.

**Comment by nazar-pc** [comment]
Will look, there are 3 PRs in my list, sorry for delays

File: worker/src/RTC/SvcConsumer.cpp
Comment (ibc) [inline worker/src/RTC/SvcConsumer.cpp:118]: I like using `std::addressof(rtpStreams)` instead of C style `&rtpStreams`, but it's not critical.

**Review by ibc — APPROVED** [review]

File: node/src/Consumer.ts
Comment (nazar-pc) [inline node/src/Consumer.ts]: Same as in producer PR, I think error should be printed and not swallowed in such cases, might be hard to debug otherwise.

File: node/src/Consumer.ts
Comment (nazar-pc) [inline node/src/Consumer.ts:1159]: The number of `!` looks worrying to me, something feels off if we need it so much in this function

File: rust/src/data_structures.rs
Comment (nazar-pc) [inline rust/src/data_structures.rs:1121]: When do you plan to do this?

File: rust/src/router/consumer.rs
Comment (nazar-pc) [inline rust/src/router/consumer.rs:641]: New line between structs/impls, please (here and in other places):
```suggestion
}

```

File: rust/tests/integration/consumer.rs
Comment (nazar-pc) [inline rust/tests/integration/consumer.rs]: Hm, why were all these removed? I recall Producer and consumer data structures were distinct for a reason.

File: rust/tests/integration/consumer.rs
Comment (nazar-pc) [inline rust/tests/integration/consumer.rs]: Same question as in producer PR: why was this change necessary?

File: node/src/Consumer.ts
Comment (jmillan) [reply node/src/Consumer.ts:1159]: For simulcast and SVC we know these have values coming from the worker, this is the reason of using `!` here. For `SimpleConsumer` they are not being populated and we don't even access them.

File: rust/src/data_structures.rs
Comment (jmillan) [reply rust/src/data_structures.rs:1121]: I've added a TODO item in the flatbuffers PR. It must be done for sure before merging that branch into v3.

File: rust/tests/integration/consumer.rs
Comment (jmillan) [reply rust/tests/integration/consumer.rs]: It was not necessary and I'll put it back to `None`. So when the value coming from the server is `false` it will be `None` rather than `Some(false)`. I find the field type `Option<bool>` confusing in itself. Doesn't it make more sense for that such member `dtx` to just be `bool` and be defaulted to `false`?

Anyway I'm putting it back to `None`.

File: rust/tests/integration/consumer.rs
Comment (jmillan) [reply rust/tests/integration/consumer.rs]: Nop, they are indeed the same. Verified. Those removed ones are represented by `scalability_mode`.

File: rust/tests/integration/consumer.rs
Comment (nazar-pc) [reply rust/tests/integration/consumer.rs]: `None` basically means it wasn't specified, use the default. We can make it just `bool`, but since it was optional I made it `Option`. While here `false` is the default, for some options the default for `None` is `true`.

**Pull request merged**
